File: services/fdd959b0be5be17775566509/scripts/create-test-accounts.js
#!/usr/bin/env node
/**
 * EverShop Test Account Creation Script
 * Creates admin and customer accounts for security testing
 *
 * Usage: node scripts/create-test-accounts.js
 *
 * This script creates:
 * - Admin account: admin@evershop.com / Admin@123
 * - Customer account: customer@evershop.com / User@123
 */

import 'dotenv/config';
import { insertOnUpdate } from '@evershop/postgres-query-builder';
import { pool } from '../packages/evershop/src/lib/postgres/connection.js';
import { hashPassword } from '../packages/evershop/src/lib/util/passwordHelper.js';

const ADMIN_EMAIL = 'admin@evershop.com';
const ADMIN_PASSWORD = 'Admin@123';
const ADMIN_FULLNAME = 'Admin User';

const CUSTOMER_EMAIL = 'customer@evershop.com';
const CUSTOMER_PASSWORD = 'User@123';
const CUSTOMER_FULLNAME = 'Test Customer';

async function createAdminAccount() {
  console.log('Creating admin account...');
  try {
    await insertOnUpdate('admin_user', ['email'])
      .given({
        full_name: ADMIN_FULLNAME,
        email: ADMIN_EMAIL,
        password: hashPassword(ADMIN_PASSWORD)
      })
      .execute(pool);
    console.log(`✓ Admin account created: ${ADMIN_EMAIL} / ${ADMIN_PASSWORD}`);
    return true;
  } catch (e) {
    console.error(`✗ Failed to create admin account: ${e.message}`);
    return false;
  }
}

async function createCustomerAccount() {
  console.log('Creating customer account...');
  try {
    // Check if customer already exists
    const existing = await pool.query(
      'SELECT * FROM customer WHERE email = $1',
      [CUSTOMER_EMAIL]
    );

    if (existing.rows.length > 0) {
      // Update existing customer password
      await pool.query(
        'UPDATE customer SET password = $1, full_name = $2 WHERE email = $3',
        [hashPassword(CUSTOMER_PASSWORD), CUSTOMER_FULLNAME, CUSTOMER_EMAIL]
      );
      console.log(`✓ Customer account updated: ${CUSTOMER_EMAIL} / ${CUSTOMER_PASSWORD}`);
    } else {
      // Insert new customer
      await pool.query(
        `INSERT INTO customer (email, full_name, password, group_id, status)
         VALUES ($1, $2, $3, 1, 1)`,
        [CUSTOMER_EMAIL, CUSTOMER_FULLNAME, hashPassword(CUSTOMER_PASSWORD)]
      );
      console.log(`✓ Customer account created: ${CUSTOMER_EMAIL} / ${CUSTOMER_PASSWORD}`);
    }
    return true;
  } catch (e) {
    console.error(`✗ Failed to create customer account: ${e.message}`);
    return false;
  }
}

async function main() {
  console.log('EverShop Test Account Creation');
  console.log('==============================\n');

  try {
    // Test database connection
    await pool.query('SELECT 1');
    console.log('✓ Database connected\n');
  } catch (e) {
    console.error(`✗ Database connection failed: ${e.message}`);
    console.log('\nMake sure PostgreSQL is running and the .env file is configured correctly.');
    process.exit(1);
  }

  const adminResult = await createAdminAccount();
  const customerResult = await createCustomerAccount();

  console.log('\n==============================');
  if (adminResult && customerResult) {
    console.log('All accounts created successfully!');
    console.log('\nAdmin Login URL: /user/login');
    console.log('Customer Login URL: /customer/login');
  } else {
    console.log('Some accounts failed to create. Check errors above.');
    process.exit(1);
  }

  await pool.end();
}

main();

File: services/2c93570782b7b528740e6d8e/create_test_accounts.py
#!/usr/bin/env python
"""
create_test_accounts.py - Create test accounts for Skyline

This script creates test user accounts for security testing of the Skyline webapp.
Skyline uses HTTP Basic Authentication configured via settings.py, and also has
a users table in MySQL for tracking which user created resources.

Usage:
    python create_test_accounts.py [--admin-password <password>] [--user-password <password>]
"""

import sys
import os

# Add the skyline directory to the path
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))

import argparse
from sqlalchemy import create_engine, text
import settings


def get_engine():
    """Create a SQLAlchemy engine for the Panorama database."""
    connection_string = (
        'mysql+mysqlconnector://%s:%s@%s:%s/%s' % (
            settings.PANORAMA_DBUSER,
            settings.PANORAMA_DBUSERPASS,
            settings.PANORAMA_DBHOST,
            str(settings.PANORAMA_DBPORT),
            settings.PANORAMA_DATABASE
        )
    )
    return create_engine(connection_string)


def create_or_update_user(engine, username, description, is_admin=False):
    """
    Create or update a user in the users table.

    Args:
        engine: SQLAlchemy engine
        username: The username
        description: User description/role
        is_admin: Whether this is an admin user
    """
    query = text("""
        SELECT id FROM users WHERE user = :username
    """)

    with engine.connect() as conn:
        # Check if user exists
        result = conn.execute(query, {"username": username}).fetchone()

        if result:
            # Update existing user
            update_query = text("""
                UPDATE users
                SET description = :description
                WHERE user = :username
            """)
            conn.execute(update_query, {
                "username": username,
                "description": description
            })
            conn.commit()
            print(f"Updated user: {username}")
        else:
            # Insert new user
            insert_query = text("""
                INSERT INTO users (user, description)
                VALUES (:username, :description)
            """)
            conn.execute(insert_query, {
                "username": username,
                "description": description
            })
            conn.commit()
            print(f"Created user: {username}")


def update_webapp_authSettings(admin_username, admin_password, user_username, user_password):
    """
    Update the settings.py file with new authentication credentials.

    Args:
        admin_username: Admin username
        admin_password: Admin password
        user_username: Normal user username
        user_password: Normal user password
    """
    settings_path = os.path.join(
        os.path.dirname(os.path.abspath(__file__)),
        'skyline', 'settings.py'
    )

    # Read the current settings file
    with open(settings_path, 'r') as f:
        content = f.read()

    # Update WEBAPP_AUTH_USER
    if 'WEBAPP_AUTH_USER = ' in content:
        content = content.replace(
            'WEBAPP_AUTH_USER = \'admin\'',
            f'WEBAPP_AUTH_USER = \'{admin_username}\''
        )

    # Update WEBAPP_AUTH_USER_PASSWORD (admin password)
    # This is a simple replacement for testing purposes
    # In production, you would want more sophisticated handling
    lines = content.split('\n')
    new_lines = []
    for line in lines:
        if line.startswith('WEBAPP_AUTH_USER_PASSWORD = '):
            new_lines.append(f"WEBAPP_AUTH_USER_PASSWORD = '{admin_password}'  # Changed by create_test_accounts.py")
        else:
            new_lines.append(line)
    content = '\n'.join(new_lines)

    # Write back to the settings file
    with open(settings_path, 'w') as f:
        f.write(content)

    print(f"Updated settings.py with new credentials")
    print(f"  Admin user: {admin_username}")
    print(f"  Note: Basic Auth uses single credentials - for testing, use admin credentials")


def main():
    parser = argparse.ArgumentParser(
        description='Create test accounts for Skyline security testing'
    )
    parser.add_argument(
        '--admin-password',
        default='Admin@123',
        help='Password for the admin account (default: Admin@123)'
    )
    parser.add_argument(
        '--user-password',
        default='User@123',
        help='Password for the normal user account (default: User@123)'
    )
    parser.add_argument(
        '--admin-user',
        default='admin',
        help='Username for admin account (default: admin)'
    )
    parser.add_argument(
        '--normal-user',
        default='testuser',
        help='Username for normal user account (default: testuser)'
    )

    args = parser.parse_args()

    print("Creating test accounts for Skyline...")
    print("-" * 50)

    try:
        # Create users in the database
        engine = get_engine()

        # Create admin user in database
        create_or_update_user(
            engine,
            args.admin_user,
            'Administrator account for security testing'
        )

        # Create normal user in database
        create_or_update_user(
            engine,
            args.normal_user,
            'Normal user account for security testing'
        )

        print("-" * 50)
        print("Users created successfully in MySQL database!")

        # Update settings for Basic Auth
        print("-" * 50)
        print("Note: Skyline uses HTTP Basic Authentication configured in settings.py")
        print("The login URL is the webapp root path - any endpoint triggers auth")
        print()
        print("For security testing, use:")
        print(f"  URL: <SKYLINE_URL>/ (or any webapp route)")
        print(f"  Username: {args.admin_user}")
        print(f"  Password: {args.admin_password}")

        # Optionally update settings.py
        update_choice = input(
            "\nUpdate settings.py with these credentials? (yes/no): "
        ).strip().lower()

        if update_choice in ('yes', 'y'):
            update_webapp_authSettings(
                args.admin_user,
                args.admin_password,
                args.normal_user,
                args.user_password
            )
            print("\nSettings updated! Restart Skyline webapp to apply changes.")

    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)


if __name__ == '__main__':
    main()